### PR TITLE
Abstract onboarding plugin code into a PluginsHelper class

### PR DESCRIPTION
**Pull request by jconroy**
The Onboarding Feature has some existing plugin functionality for retrieving the active and installed plugins that could be useful to other features (e.g. Marketing Dashboard which also allows installation/activation of plugins).

This PR abstracts out the onboarding code into a PluginsHelper class and adds some additional helpers:
* is_plugin_active()
* is_plugin_installed()
* get_plugin_data()
* get_plugin_path_from_slug()

### Detailed test instructions:

I've added some initial tests in b69652a and the existing [Onboarding Plugins tests](https://github.com/woocommerce/woocommerce-admin/blob/master/tests/api/onboarding-plugins.php) provide some coverage as well.

For manual testing i've put together simple plugin you can use https://gist.github.com/jconroy/daa9751d094dc0a311526f85b5f2c580 to check output of the various functions based on what you are running in your dev environment.

cc @danielbitzer (was most of your work, I'm just spinning it out for review)

### CODE REVIEW

File: tests/plugins-helper.php [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		// Phpunit test environment active plugins option is empty.
+		$this->assertEquals( array( 'facebook-for-woocommerce' ), $active_slugs, 'Facebook for WooCommerce should be listed as active.' );
+	}
+
+	/**
+	 * Test is_plugin_installed()
+	 */
+	public function test_is_plugin_installed() {
+
+		// WooCommerce is installed in the test envrionment.
```
Comment (becdetat) [inline tests/plugins-helper.php]: Typo here

File: tests/plugins-helper.php [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		$installed = PluginsHelper::is_plugin_installed( 'woocommerce' );
+		$this->assertEquals( true, $installed, 'WooCommerce should be installed.' );
+
+		// Invalud plugin is not.
```
Comment (becdetat) [inline tests/plugins-helper.php]: Typo here

File: tests/plugins-helper.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	public function test_is_plugin_active() {
+
+		// Check if facebook is not active. Phpunit test environment active plugins option is empty.
+		$active = PluginsHelper::is_plugin_active( 'facebook-for-woocommerce' );
+		$this->assertEquals( false, $active, 'Should not be any active slugs.' );
+
+		// Get facebook plugin path.
+		$fb_path = PluginsHelper::get_plugin_path_from_slug( 'facebook-for-woocommerce' );
+
+		// Activate facebookn plugin.
```
Comment (becdetat) [inline tests/plugins-helper.php]: Typo here

File: tests/plugins-helper.php (hunk 1 above)
Comment (jconroy) [reply tests/plugins-helper.php]: fixed

File: tests/plugins-helper.php (hunk 2 above)
Comment (jconroy) [reply tests/plugins-helper.php]: fixed

File: tests/plugins-helper.php (hunk 3 above)
Comment (jconroy) [reply tests/plugins-helper.php]: fixed

**Pull request merged**